### PR TITLE
Feature/1.18 chunks update

### PR DESCRIPTION
**Pull request by Tides**
This is for you guys to review and comment on the changes I made to adapt to the new 1.18 format and for it to be used in previous versions as well

### CODE REVIEW

File: Obsidian/ChunkData/BiomeContainer.cs [hunk 1]
```diff
@@ -1,42 +1,71 @@
...
+        this.Palette = bitsPerEntry.DetermineBiomePalette();
+
+        this.DataArray = new(bitsPerEntry, 64);
     }
 
-    public void SetBiome(int bx, int by, int bz, Biomes biome)
+    public bool Set(int x, int y, int z, Biomes biome)
     {
-        Biomes[GetIndex(bx, by, bz)] = (int)biome;
+        y %= 16;
```
Comment (Jonpro03) [inline Obsidian/ChunkData/BiomeContainer.cs]: C# does modulo on negatives differently than Java. You'll want to use NumericHelpers.Modulo here if you expect y will ever be negative.

Comment (Tides) [reply Obsidian/ChunkData/BiomeContainer.cs]: 👍🏼

File: Obsidian/ChunkData/BiomeContainer.cs [hunk 2]
```diff
@@ -1,42 +1,73 @@
...
-    public Biomes GetBiome(int bx, int by, int bz)
+    public Biomes Get(int x, int y, int z)
     {
-        return (Biomes)Biomes[GetIndex(bx, by, bz)];
+        y = NumericsHelper.Modulo(y, 16);
+
+        var storageId = this.DataArray[this.GetIndex(x, y, z)];
+
+        return this.Palette.GetBiomeFromIndex(storageId);
     }
```
Comment (Craftplacer) [inline Obsidian/ChunkData/BiomeContainer.cs:44]: Implement https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/

File: Obsidian/ChunkData/BlockStateContainer.cs [hunk 3]
```diff
@@ -3,63 +3,79 @@
...
+    public Block Get(int x, int y, int z)
     {
-        y %= 16;
-        int storageId = this.BlockStorage[GetIndex(x, y, z)];
+        y = NumericsHelper.Modulo(y, 16);
+
+        int storageId = this.DataArray[GetIndex(x, y, z)];
 
         return this.Palette.GetStateFromIndex(storageId);
     }
```
Comment (Craftplacer) [inline Obsidian/ChunkData/BlockStateContainer.cs:45]: Same as previous comment

**Review by Craftplacer — CHANGES_REQUESTED** [review]
Looks ok, but a bit more usage of Indexers could be cool.

**Pull request merged**